### PR TITLE
Current stage tests

### PR DESCRIPTION
**Pull request by davluangu**
@robertzk 
1) added some tests, 
2) upped version and tagged `0.5.1`

By the way, to run `devtools::test()` on this package, I had to `install_github('robertzk/objectdiff')`:
```R
> devtools::test('~/dev/stagerunner/')
Loading stagerunner
Testing stagerunner
boolean_fill : ..............
caching on a stageRunner object : ..................
stageRunner coalescing : 1.........Error in library(objectdiff) : there is no package called ‘objectdiff’
```
Using the package doesn't have a dependency on it, but running the tests does. Should it be added to the package dependency list? What is the proper protocol for this?

All tests pass locally:
```R
> devtools::test('~/dev/stagerunner/')
Loading stagerunner
Testing stagerunner
boolean_fill : ..............
caching on a stageRunner object : ..................
stageRunner coalescing : ........................
compare_stage_keys : .................
contains_true : ............
copy_env : ...
is_pre_stagerunner : ........
nested names : .
normalize_stage_keys : .............
stageRunner overlaying : ...........
printing stageRunner : .........
printing stageRunner with caching : .........
serialization : ....
special_and_lists function : .................
special_or_lists function : ..................
stageRunnerNode$around : ...
stageRunner$around : ..........
stagerunner$has_key : .....
stageRunner mode parameter : .........
stagerunner$transform : ..
verbose runs : .....
stageRunnerNode : ..
stageRunner : ..................................................................................
tracked_environment : ..................
treeSkeleton : .......
utils : ...

DONE
```

### CODE REVIEW

**Comment by robertzk** [comment]
It's already in the [Suggests](https://github.com/robertzk/stagerunner/blob/master/DESCRIPTION#L22).

**Comment by robertzk** [comment]
Will merge when CI agrees.

File: tests/testthat/test-stage_runner.r [hunk 1]
```diff
@@ -378,4 +378,56 @@ test_that('it correctly figures out the next stage in a stageRunner with deeply
...
+    info = 'this stageRunner should not execute the last stage')
+  sr$run('2/2')
+  expect_identical(sr$current_stage(), '2/2',
+    info = 'this stageRunner should claim to be completely done')
+})
+
+test_that('it correctly figures out the current stage in a stageRunner with deeply nested executed stages', {
+  sr <- stageRunner(new.env(),
+    list(a = force, b = list(c = list(force, d = list(force, force, force))), force))
+                                # this stage should be marked as next ^
```
Comment (robertzk) [inline tests/testthat/test-stage_runner.r]: Nice comment!

Comment (davluangu) [inline tests/testthat/test-stage_runner.r]: oops, actually that is a copy and paste of your comment from the `next_stage` tests.
I want to change it, hold on.

**Comment by davluangu** [comment]
@robertzk Travis is pleased :shipit:

**Pull request merged**